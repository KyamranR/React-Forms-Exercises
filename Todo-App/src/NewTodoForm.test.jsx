import { render, fireEvent } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";
import { vi } from "vitest";

test("renders without crashing", () => {
  render(<NewTodoForm addTodo={() => {}} />);
});

test("matches snapshot", () => {
  const { asFragment } = render(<NewTodoForm addTodo={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});

test("can add a new todo", () => {
  const addTodo = vi.fn();
  const { getByText, getByPlaceholderText } = render(
    <NewTodoForm addTodo={addTodo} />
  );
  const input = getByPlaceholderText("New task");
  const button = getByText("Add Todo");

  fireEvent.change(input, { target: { value: "Test Todo" } });
  fireEvent.click(button);
  expect(addTodo).toHaveBeenCalledWith("Test Todo");
});
