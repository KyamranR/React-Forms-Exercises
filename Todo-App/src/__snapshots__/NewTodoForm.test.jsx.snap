// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <input
      placeholder="New task"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
